File: mobile/ios/Air/SubModules/WalletCore/Models/MAutolockOption.swift

import UIKit
import WalletContext


public enum MAutolockOption: String, Equatable, Hashable, Codable, Sendable, CaseIterable, Identifiable {
    case never = "never"
    case thirtySeconds = "1"
    case threeMinutes = "2"
    case tenMinutes = "3"

    public var period: TimeInterval? {
        switch self {
        case .never: return nil
        case .thirtySeconds: return 30
        case .threeMinutes: return 3 * 60
        case .tenMinutes: return 10 * 60
        }
    }

    public var displayName: String {
        switch self {
        case .never: return "Never"
        case .thirtySeconds: return "30 seconds"
        case .threeMinutes: return "3 minutes"
        case .tenMinutes: return "10 minutes"
        }
    }
    
    public var id: Self { self }
}

File: mobile/android/air/SubModules/WalletContext/src/main/java/org/mytonwallet/app_air/walletcontext/models/MAutoLockOption.kt
package org.mytonwallet.app_air.walletcontext.models

import org.mytonwallet.app_air.walletcontext.R
import org.mytonwallet.app_air.walletcontext.helpers.LocaleController

enum class MAutoLockOption(val value: String, val period: Int?) {
    NEVER("never", null),
    THIRTY_SECONDS("1", 30),
    THREE_MINUTES("2", 3 * 60),
    TEN_MINUTES("3", 10 * 60);

    companion object {
        fun fromValue(value: String?): MAutoLockOption? {
            if (value == null)
                return NEVER
            return entries.firstOrNull { it.value == value }
        }
    }

    val displayName: String
        get() {
            return when (this) {
                NEVER -> {
                    LocaleController.getString(R.string.Security_AutoLock_Never)
                }

                THIRTY_SECONDS -> {
                    LocaleController.getString(R.string.Security_AutoLock_Seconds, listOf("30"))
                }

                THREE_MINUTES -> {
                    LocaleController.getString(R.string.Security_AutoLock_Minutes, listOf("3"))
                }

                TEN_MINUTES -> {
                    LocaleController.getString(R.string.Security_AutoLock_Minutes, listOf("10"))
                }
            }
        }
}
